eslint-plugin-prettier.js

```diff
@@ -45,15 +45,45 @@ let prettier;
  * @returns {Object} An object containing numeric `line` and `column` keys.
  */
 function getLocFromIndex(context, index) {
-  // If `sourceCode.getLocFromIndex` is available from ESLint, use it.
-  // Otherwise, use the private version from eslint/lib/ast-utils.
-  // `sourceCode.getLocFromIndex` was added in ESLint 3.16.0.
+  // If `sourceCode.getLocFromIndex` is available from ESLint, use it - added
+  // in ESLint 3.16.0.
   const sourceCode = context.getSourceCode();
   if (typeof sourceCode.getLocFromIndex === 'function') {
     return sourceCode.getLocFromIndex(index);
   }
+  // Otherwise, try to use the private version from eslint/lib/ast-utils -
+  // added in ESLint 3.11.0.
   const astUtils = require('eslint/lib/ast-utils');
-  return astUtils.getLocationFromRangeIndex(sourceCode, index);
+  if (typeof astUtils.getLocationFromRangeIndex === 'function') {
+    return astUtils.getLocationFromRangeIndex(sourceCode, index);
+  }
+  // `getLocFromIndex` and `getLocationFromRangeIndex` are preferred because
+  // they use a line index cache, so lastly use this unoptimized version.
+  const text = sourceCode.getText();
+  const lines = sourceCode.getLines();
+  if (typeof index !== 'number') {
+    throw new TypeError('Expected `index` to be a number.');
+  }
+  if (index < 0 || index > text.length) {
+    throw new RangeError('Index out of range.');
+  }
+  if (index === text.length) {
+    return {
+      line: lines.length,
+      column: lines[lines.length - 1].length
+    };
+  }
+  let offset = 0;
+  for (let i = 0; i < lines.length; i++) {
+    if (index >= offset && index <= offset + lines[i].length) {
+      return {
+        line: i + 1,
+        column: index - offset
+      };
+    }
+    // Add 1 for the line ending character
+    offset = offset + lines[i].length + 1;
```

Comment:
I think this will be incorrect if a file contains CRLF newlines, because the line ending will have a length of 2.

IIRC, there isn't enough information in the `lines` array to compute this accurately because it doesn't store which linebreaks were used. `astUtils.getLocationFromRangeIndex` was implemented by [splitting the text into lines again](https://github.com/eslint/eslint/blob/18a519fa2971e54a639e2b3320aeecd9a31b390a/lib/ast-utils.js#L408-L438) and including linebreaks in the result.